packages/deployments/contracts/contracts/core/connext/facets/RoutersFacet.sol

```diff
@@ -563,7 +568,18 @@ contract RoutersFacet is BaseConnextFacet {
     if (_amount == 0) revert RoutersFacet__addLiquidityForRouter_amountIsZero();
 
     // Get the canonical asset ID from the representation.
-    (TokenId memory canonical, bytes32 key) = _getApprovedCanonicalId(_local);
+    // NOTE: not using `_getApprovedCanonicalId` because candidate can *only* be local
+    TokenId memory canonical = s.representationToCanonical[_local];
+    if (canonical.domain == 0) {
+      // Assume you are on the canonical domain, which does not update the above mapping
+      // If this is an incorrect assumption, the approval should fail
+      canonical.domain = s.domain;
+      canonical.id = TypeCasts.addressToBytes32(_local);
+    }
+    bytes32 key = AssetLogic.calculateCanonicalHash(canonical.id, canonical.domain);
+    if (!_isAssetWhitelistRemoved() && !s.approvedAssets[key]) {
```

Comment:
If `_isAssetWhitelistRemoved() == true`, then for assets that have not been added to the protocol we are assuming:

1. The canonical domain is the current domain
2. The canonical id is basically the `_local` provided

And then using these incorrect values, we calculate `key` and allow routers to deposit liquidity and finally emit `RouterLiquidityAdded` using the incorrect `key`.